moveit_ros/planning/planning_pipeline/include/moveit/planning_pipeline/planning_pipeline.h

```diff
@@ -173,9 +173,18 @@ class MOVEIT_PLANNING_PIPELINE_EXPORT PlanningPipeline
     return robot_model_;
   }
 
+  /** \brief Get current status of the planning pipeline */
+  bool isActive() const
+  {
+    return active_;
+  }
+
 private:
   void configure();
 
+  // Flag that indicates whether or not the planning pipeline is currently solving a planning problem
+  std::atomic<bool> active_;
```

Comment:
Why is this atomic? I don't see any atomic operations

Reply:
I made it atomic since I assumed it might be possible that a thread tries to read this variable through `isActive()` while the another thread is calling `generatePlan(...)` which writes to it. Maybe I am misunderstanding something here: Is it possible to have a race condition between two member functions which access the same member variable?

Reply:
It looked like it would only be used by the same thread to me. If it's accessed by multiple threads, this makes sense.